sourcegraph/about
# HubSpot Master Forms

This closes #5409 and consolidates most of our HubSpot forms under 4 master forms that map to specific `formId`s. Some forms that do not fall under the master forms still use the specific `formId` syntax in the API.

Feel free to take a look at our [Master List](https://docs.google.com/spreadsheets/d/1T7yPZJ3e82-J4IrasuFYCIi1-_MowxRqAWzZ0gLEUQ8/edit#gid=0) to understand the mapping and also the [tech requirements](https://docs.google.com/document/d/1qa393uAW_tjwNtH9jHf_d__PCJgmbsxNoxi9cO4XOcE/edit).

### Changelog
- Converted our HubSpot hook into a component for convenient rendering in markup for both MDX and non-MDX pages. This means the old `EmbeddedHubSpot` component is now deprecated in favour of one component to rule them all. The new API is much more simplistic for both cases. I will follow up with a PR to update the [handbook section](https://handbook.sourcegraph.com/departments/marketing/blog/#hubspot-form-embed-code) for embedded HubSpot forms after this PR is merged.
- Added a `useLandingSource` hook to grab and store the initial landing source of a user in `sessionStorage`. This will be used to track landing sources in our HubSpot form submission data.
- Consolidated all our contact pages in a more convenient template mapped index since they all use very similar props, content, and forms. This deprecates a lot of duplicated code into one template so we can handle contact page redesigns quicker in the future.
- Cleaned up some broken links on our abcs ebook page
- Updated our master forms in HubSpot to be more responsive

### Test

1. Ensure prettier has standardized the proposed changes.
2. Ensure every page that has a form renders the correct form
3. Ensure every contact page renders correctly
4. Start a new session at a random page (landing source) and navigate to a page with a form. Ensure that the `first_source_url` hidden form field is populated with the landing source. Also ensure that the `form_submission_source` is populated with the page you're submitting the form on. Submit a test submission and ensure both pieces of data are sent and captured in your form submission on HubSpot.

### TODO
- We are awaiting a few more requirements. This is blocked/paused until `June 22, 2022`
- Two redirects need to be created in HubSpot after this PR is merged
  1. https://info.sourcegraph.com/talk-to-a-developer -> `/contact/talk-to-a-developer`, and;
  2. https://info.sourcegraph.com/team-pricing -> `/contact/team-pricing`
